File: R/primes/sieve.R
go <- function(n) {
  if (n <= 2L) return(2L)
  nums <- seq.int(2, n)
  N <- as.integer(ceiling(sqrt(n)))
  for (i in 2L:N) {
    nums <- nums[!(nums %% i == 0L)]
  }
  return(c(nums, go(N)))
}

system.time(go(10L))
system.time(go(100L))
system.time(go(1000L))
system.time(go(10000L))
system.time(go(100000L))
system.time(go(1000000L))

File: julia/primes/sieve.jl
module Sieve

function go(x::Int)::Array{Int}
  if x <= 2
    return [2]
  end
  nums::Array{Int} = 2:x
  rootx::Int = ceil(sqrt(x))
  for n::Int = 2:rootx
    filter!(x -> x % n > 0, nums)
  end
  return vcat(nums, go(rootx))
end

end

@time Sieve.go(10)
@time Sieve.go(100)
@time Sieve.go(1000)
@time Sieve.go(10000)
@time Sieve.go(100000)
@time Sieve.go(1000000)
